gprofiler/main.py

```diff
@@ -228,34 +233,34 @@ def start(self):
 
                 # others - are ignored, with a warning.
                 logger.warning(f"Failed to start {prof.__class__.__name__}, continuing without it", exc_info=True)
-                self.process_profilers.remove(prof)
+                self.process_profilers.remove(cast(ProcessProfilerBase, prof))
 
-    def stop(self):
+    def stop(self) -> None:
         logger.info("Stopping ...")
         self._stop_event.set()
         self._system_metrics_monitor.stop()
 
         for prof in self.all_profilers:
             prof.stop()
 
-    def _snapshot(self):
+    def _snapshot(self) -> None:
         local_start_time = datetime.datetime.utcnow()
         monotonic_start_time = time.monotonic()
         process_profilers_futures = []
         for prof in self.process_profilers:
             prof_future = self._executor.submit(prof.snapshot)
-            prof_future.name = prof.name
+            prof_future.name = prof.name  # type: ignore
```

Comment:
`cast`?

Reply:
Ahh, it's due to `prof_future.name` missing. Indeed a hack... lol. Can comment that next to the type ignore?

Reply:
Indeed a hack